### PR TITLE
CB: 11982  -Created new cordova config command

### PR DESCRIPTION
**Pull request by audreyso**
* Not ready to review yet. Thanks!

### What does this PR do?
Created new cordova config command (edit, ls, delete, get, set)
### Checklist
- [X] [Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database
- [X] Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected.
- [X] Added automated test coverage as appropriate for this change.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
     "configstore": "^2.1.0",
     "cordova-common": "2.0.0",
     "cordova-lib": "6.5.0",
+    "editor": "^1.0.0",
```
Comment (audreyso) [inline package.json]: will remove

File: doc/config.txt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+--get <key> ...................................... Echo the config value to 																		   stdout.
+--delete <key> ................................... Deletes the key from all 																		   configuration files.
+--edit ........................................... Opens the config file in an 																		   editor.
+
+Example
+
+cordova-cli config set <key> <value> --> cordova config set autosave true
+cordova-cli config get <key> 		 --> cordova config get autosave
+cordova-cli config delete <key>		 --> cordova config delete autosave
+cordova-cli config edit 			 --> cordova config edit
```
Comment (audreyso) [inline doc/config.txt]: will need to add in "ls" command

File: src/cli.js [hunk 3]
```diff
@@ -160,7 +183,7 @@ module.exports = function (inputArgs, cb) {
 };
 
 function getSubCommand(args, cmd) {
-    if(cmd === 'platform' || cmd === 'platforms' || cmd === 'plugin' || cmd === 'plugins' || cmd === 'telemetry') {
+    if(cmd === 'platform' || cmd === 'platforms' || cmd === 'plugin' || cmd === 'plugins' || cmd === 'telemetry' || cmd === 'config') {
```
Comment (purplecabbage) [inline src/cli.js]: This is absolutely going to be a merge conflict with #270 
https://github.com/apache/cordova-cli/pull/270/files#diff-d9d9214113dea1c364abd4ba3c70be11R317

**Comment by purplecabbage** [comment]
This PR will need to follow #270 and will need to be rebased, and have a conflict fix.

**Comment by stevengill** [comment]
Can you also update doc/cordova.txt with info about this new command?

We will also want to update cordova-docs repo with this new command before cordova-7 is released

File: doc/cordova.txt [hunk 4]
```diff
@@ -6,6 +6,7 @@ Global Commands
     create ............................. Create a project
     help ............................... Get help for a command
     telemetry .......................... Turn telemetry collection on or off
+    config ............................. Store environment variables
```
Comment (audreyso) [inline doc/cordova.txt]: @stevengill Is this what you had in mind for cordova.txt? More details needed?

File: doc/config.txt [hunk 5]
```diff
@@ -0,0 +1,22 @@
+Synopsis
+
+    cordova-cli config <command> [options]
+    
+
+The config command can be used to set, get, delete, edit, and list the 
+contents of the user files.
```
Comment (filmaj) [inline doc/config.txt]: "user files" here was confusing to me - it was unclear to me what that is.

Based on the [cordova-discuss proposal](https://github.com/cordova/cordova-discuss/pull/63/commits/788be4b3a2f360cdc2c5b4ddd46d7547a7f45416), we want to model this command after `npm config`, and its synopsis goes like this:

>        npm-config - Manage the npm configuration files

`git config` serves a similar purpose, I think, and its synopsis is:

>        git-config - Get and set repository or global options

One suggestion: "set, get, delete, edit, and list global cordova options" ?

File: doc/cordova.txt [hunk 6]
```diff
@@ -6,6 +6,8 @@ Global Commands
     create ............................. Create a project
     help ............................... Get help for a command
     telemetry .......................... Turn telemetry collection on or off
+    config ............................. Store and retrieve environment variables
```
Comment (filmaj) [inline doc/cordova.txt]: Along the same lines as my previous comment, to me, environment variables means something other than program-specifc (i.e. cordova-specific) configurations. Whatever wording we choose for the synopsis, I would also update this portion to reflect that.

**Comment by audreyso** [comment]
@filmaj Thanks for your help! :)

File: doc/config.txt (hunk 2 above)
Comment (stevengill) [inline doc/config.txt]: is tabbing of the arrows a little off here?

**Comment by stevengill** [comment]
this has been merged, can you close it @audreyso

**Pull request closed without merging**